vdev/src/commands/build/publish_metadata.rs

```diff
@@ -19,19 +19,19 @@ pub struct Cli {}
 impl Cli {
     pub fn exec(self) -> Result<()> {
         // Generate the Vector version and build description.
-        let version = util::read_version()?;
+        let version = env::var("VERSION").or_else(|_| util::read_version())?;
```

Comment:
It stuck out to me that this only reads `VERSION` while `app.rs` also reads `VECTOR_VERSION`. Is there a reason they are different? Should there maybe be a helper function they both use that reads both along with the fallback to reading it from `git`?

Reply:
Good spot. The publish metadata command is run (AFAICT) only at the beginning of the `publish.yml` workflow, and this workflow is what provides `VECTOR_VERSION` , so I don't think it would improve anything functionally, by adding it here too (or in a central helper function).
But I guess it wouldn't hurt to do that either (at least in this current state of the workflows), if it would dispel confusion.
I guess an alternative is adding a code comment.

What do you think makes the most sense?

Reply:
Hmm, yeah, I could just see it causing confusion. At the least, a code comment would help.

Reply:
👍 I extracted that into a helper.